src/TestProblem/Hydro/ClusterMerger/Par_Init_ByFunction_ClusterMerger.cpp

```diff
@@ -488,9 +586,429 @@ void Read_Particles_ClusterMerger( std::string filename, long offset, long num,
+
+         for (int c=0; c<Merger_Coll_NumHalos; c++) {
+            delete[] ParX_sum[c];
+            delete[] ParY_sum[c];
+            delete[] ParZ_sum[c];
+            delete[] ParM_sum[c];
+            delete[] VelX_sum[c];
+            delete[] VelY_sum[c];
+            delete[] VelZ_sum[c];
+         }
+         delete[] ParX_sum;
+         delete[] ParY_sum;
+         delete[] ParZ_sum;
+         delete[] ParM_sum;
+         delete[] VelX_sum;
+         delete[] VelY_sum;
+         delete[] VelZ_sum;
+         for (int c=0; c<Merger_Coll_NumHalos; c++) {
+            delete[] ParX[c];
+            delete[] ParY[c];
+            delete[] ParZ[c];
+            delete[] ParM[c];
+            delete[] VelX[c];
+            delete[] VelY[c];
+            delete[] VelZ[c];
+         }
+         delete[] ParX;
+         delete[] ParY;
+         delete[] ParZ;
+         delete[] ParM;
+         delete[] VelX;
+         delete[] VelY;
+         delete[] VelZ;
+      } // while ( IfConverge == false )
+   } // if ( (CurrentMaxLv  &&  AdjustPos == true) || (CurrentMaxLv  &&  AdjustVel == true) ) 
+
+
+// Find the BH particles and adjust their position and velocity
+   for (int c=0; c<Merger_Coll_NumHalos; c++) {
+      double Cen_Tmp[3] = { -__FLT_MAX__, -__FLT_MAX__, -__FLT_MAX__ };   // set to -inf
+      double Vel_Tmp[3] = { -__FLT_MAX__, -__FLT_MAX__, -__FLT_MAX__ };
+      for (long p=0; p<amr->Par->NPar_AcPlusInac; p++) {
+         if ( amr->Par->Mass[p] >= (real)0.0  &&  amr->Par->Type[p] == real(PTYPE_CEN+c) ){
+            if ( CurrentMaxLv  &&  AdjustPos == true ){
+               amr->Par->PosX[p] = min_pos[c][0];
+               amr->Par->PosY[p] = min_pos[c][1];
+               amr->Par->PosZ[p] = min_pos[c][2];    
+            }
+            if ( CurrentMaxLv  &&  AdjustVel == true ){
+               amr->Par->VelX[p] = DM_Vel[c][0];
+               amr->Par->VelY[p] = DM_Vel[c][1];
+               amr->Par->VelZ[p] = DM_Vel[c][2];        
+            }   
+            Cen_Tmp[0] = amr->Par->PosX[p];
+            Cen_Tmp[1] = amr->Par->PosY[p];
+            Cen_Tmp[2] = amr->Par->PosZ[p];
+            Vel_Tmp[0] = amr->Par->VelX[p];
+            Vel_Tmp[1] = amr->Par->VelY[p];
+            Vel_Tmp[2] = amr->Par->VelZ[p];
+            break;
+         }
+      }
+
+//    use MPI_MAX since Cen_Tmp[] is initialized as -inf
+      MPI_Allreduce( Cen_Tmp, Cen_new[c], 3, MPI_DOUBLE, MPI_MAX, MPI_COMM_WORLD );
+      MPI_Allreduce( Vel_Tmp, Cen_Vel[c], 3, MPI_DOUBLE, MPI_MAX, MPI_COMM_WORLD );
+   } // for (int c=0; c<Merger_Coll_NumHalos; c++)
+
+   if ( CurrentMaxLv  &&  AdjustPos == true ){
+      const bool TimingSendPar_No = false;
+      Par_PassParticle2Sibling( lv, TimingSendPar_No );
+      Par_PassParticle2Son_MultiPatch( lv, PAR_PASS2SON_EVOLVE, TimingSendPar_No, NULL_INT, NULL );
+   }
+   } // if ( fixBH == false )
+   else {   // if ( fixBH == true )
+      for (int d=0; d<3; d++)   Cen_new[0][d] = 7.5; 
```

Comment:
We should not do it this way. If we fix the BH position for a single cluster, it should be to the initial cluster position, wherever it is. We should not use these hard-coded values.